TypeCobol.LanguageServer/TypeCobolLanguageServer.cs

```diff
@@ -278,6 +282,48 @@ public void WholeDocumentChanged(object sender, EventArgs eventArgs)
             UpdateTokensLines(compilationDocument);
         }
 
+        /// <summary>
+        /// LSR Cfg Analyzer ID.
+        /// </summary>
+        public const string lspcfgId = "lsp-cfg";
+
+        /// <summary>
+        /// Method to update CFG/DFA information.
+        /// </summary>
+        /// <param name="fileCompiler">The underlying File Compiler</param>
+        /// <param name="lsrMode)">True if this action is executed in a LSR Mode</param>
+        /// <returns>CFG/DFA Data information</returns>
+        public CfgDfaParams UpdateCfgDfaInformation(DocumentContext docContext, bool lsrMode)
+        {
+            CfgDfaParams result = null;
+            docContext.FileCompiler.CompilationResultsForProgram.TryGetAnalyzerResult(lspcfgId, out IList<ControlFlowGraph<Node, object>> cfgs);
+            if (cfgs != null)
+            {                
+                //Create a temporary dot file.
+                string tempFile = Path.GetTempFileName();
+                using (TextWriter writer = lsrMode ? (TextWriter)new StringWriter() : File.CreateText(tempFile))
+                {
+                    CfgDfaParamsBuilder builder = new CfgDfaParamsBuilder(new TextDocumentIdentifier(docContext.TextDocument.uri), lsrMode ? null : tempFile);
```

Comment:
Maybe we could keep a builder instance and add a `Clear` or `Reset` method to it instead of creating a new builder each time the notification is called.

Reply:
Hum... At this time I don't know if we are thread safe here

Reply:
You're right